core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -760,15 +760,21 @@ object SQLPlanParser extends Logging {
         trim.replaceAll("""^\[+""", "").replaceAll("""\]+$""", "").
         replaceAll("cast\\(", "").split("windowspecdefinition").map(_.trim)
 
-    // Get functionname from each array element except the last one as it doesn't contain
+    // Get function name from each array element except the last one as it doesn't contain
     // any window function
-    for ( i <- 0 to windowExprs.size - 1 ) {
-      val windowFunc = windowFunctionPattern.findAllIn(windowExprs(i)).toList
-      val expr = windowFunc.last
-      val functionName = getFunctionName(windowFunctionPattern, expr)
-      functionName match {
-        case Some(func) => parsedExpressions += func
-        case _ => // NO OP
+    if (windowExprs.nonEmpty) {
+      // Iterate through each expression in windowExprs
+      for (expr <- windowExprs) {
```

Comment:
That's a different behavior compared to the previous code block because this new code iterates through all the entries while the previous code does not process the last element.
The comment outside the if-block matches the original behavior.

Reply:
Kept the original code where we do not process last element. But this would not go inside the loop if `windowExprs` is empty.